### PR TITLE
rgw: init oldest period after setting run_sync_thread

### PR DESCRIPTION
**Pull request by oritwas**
Fixes: http://tracker.ceph.com/issues/21996
Signed-off-by: Orit Wasserman <owasserm@redhat.com>

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -4449,20 +4449,20 @@ int RGWRados::init_complete()
...
       || current_period.get_id().empty()) {
     run_sync_thread = false;
   }
 
+  if (run_sync_thread) {
+    // initialize the log period history. we want to do this any time we're not
+    // running under radosgw-admin, so we check run_sync_thread here before
+    // disabling it based on the zone/zonegroup setup
+    meta_mgr->init_oldest_log_period();
+  }
```
Comment (cbodley) [inline src/rgw/rgw_rados.cc]: moving this down is fine, but the comment still talks about checking `run_sync_thread here before disabling it based on the zone/zonegroup setup`, which happens in the block above - could you please update or remove the comment?

**Comment by cbodley** [comment]
@oritwas i found the crash bug in `RGWPeriodHistory` - mind pulling in this fix while you're at it?
```patch
diff --git a/src/rgw/rgw_period_history.cc b/src/rgw/rgw_period_history.cc
index 895700f..1f0dbee 100644
--- a/src/rgw/rgw_period_history.cc
+++ b/src/rgw/rgw_period_history.cc
@@ -137,6 +137,8 @@ RGWPeriodHistory::Impl::Impl(CephContext* cct, Puller* puller,
 
     // get a cursor to the current period
     current_cursor = make_cursor(current_history, current_period.get_realm_epoch());
+  } else {
+    current_history = histories.end();
   }
 }
```

**Comment by oritwas** [comment]
found it too :) , will include it

File: src/rgw/rgw_rados.cc (hunk 1 above)
Comment (oritwas) [reply src/rgw/rgw_rados.cc]: sure

**Comment by oritwas** [comment]
http://pulpito.ceph.com/cbodley-2017-11-01_14:19:58-rgw-wip-rgw-21996-distro-basic-smithi/

**Pull request merged**